Unexpected CI failure

Once I pushed the #680 I saw that the CI (kokoro) build failed. And that was strange because that PR just adds a utility script which does not interfere with anything. After forcing the kokoro to run again the CI passed.

A race condition ?

The failure commit SHA is `3e11077`. The relevant fragment of the CI build log:

```
Scanning dependencies of target dram_2_32x1d_bin
[ 37%] Generating dram_4_32x1s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.net, dram_4_32x1s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/pack.log
[ 37%] Generating dram_2_64x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.net, dram_2_64x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/pack.log
[ 37%] Generating dram_1_256x1s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.net, dram_1_256x1s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/pack.log
[ 37%] Generating dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.eblif, dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_synth.v
[ 37%] Generating dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_io.place
[ 37%] Generating dram_4_32x2s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.eblif, dram_4_32x2s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_synth.v
[ 37%] Generating dram_4_32x2s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.eblif, dram_4_32x2s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_synth.v
[ 37%] Generating dram_4_32x2s/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_io.place
[ 37%] Generating dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.net, dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/pack.log

VPR FPGA Placement and Routing.
Version: 8.0.0-dev+vpr-7.0.5-7047-g9911b288b
Revision: vpr-7.0.5-7047-g9911b288b
Compiled: 2019-04-18T19:05:15 (Release build)
Compiler: GNU 7.3.0 on Linux-4.4.0-101-generic x86_64
University of Toronto
verilogtorouting.org
vtr-users@googlegroups.com
This is free open source code under MIT license.

VPR was run with the following command-line:
/tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/env/conda/bin/vpr /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/xc7/archs/artix7/devices/xc7a50t-basys3-roi-virt/arch.unique_pack.xml /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/xc7/tests/dram/dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.eblif --device xc7a50t-basys3-test --read_rr_graph /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/xc7/archs/artix7/devices/rr_graph_xc7a50t-basys3_test.rr_graph.real.xml --min_route_chan_width_hint 100 --max_criticality 0.0 --max_router_iterations 500 --routing_failure_predictor off --router_high_fanout_threshold -1 --constant_net_method route --route_chan_width 500 --clock_modeling route --place_algorithm bounding_box --enable_timing_computations off --allow_unrelated_clustering on --round_robin_prepacking on --pack

Using up to 16 parallel worker(s)

Architecture file: /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/xc7/archs/artix7/devices/xc7a50t-basys3-roi-virt/arch.unique_pack.xml
Circuit name: top

# Loading Architecture Description
Warning 1: Model 'CE_VCC' output port 'VCC' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
Warning 2: Model 'SR_GND' output port 'GND' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
Warning 3: Model 'NO_FF' input port 'D' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 4: Model 'NO_DRAM' input port 'A' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 5: Model 'RAMB18E1_VPR' input port 'ADDRBTIEHIGH' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 6: Model 'RAMB18E1_VPR' input port 'ADDRATIEHIGH' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 7: Model 'RAMB36E1_PRIM' input port 'CASCADEINB' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 8: Model 'RAMB36E1_PRIM' input port 'CASCADEINA' has no timing specification (no clock specified to create a sequential input port, not combinationally connected to any outputs, not a clock input)
Warning 9: Model 'RAMB36E1_PRIM' output port 'CASCADEOUTB' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
Warning 10: Model 'RAMB36E1_PRIM' output port 'CASCADEOUTA' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
Warning 11: Model 'VCC' output port 'VCC' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
Warning 12: Model 'GND' output port 'GND' has no timing specification (no clock specified to create a sequential output port, not combinationally connected to any inputs, not a clock output)
# Loading Architecture Description took 0.09 seconds (max_rss 14.3 MiB, delta_rss +8.4 MiB)
Warning 13: Non-zero switch output capacitance (4e-15) has no effect when switch 'buffer' is used for connection block inputs
# Building complex block graph
Warning 14: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 15: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 16: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 17: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 18: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 19: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 20: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 21: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 22: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 23: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 24: WE_MUX[0].CE[0] unconnected pin in architecture.
Warning 25: WE_MUX[0].WE[0] unconnected pin in architecture.
Warning 26: WE_MUX[0].WE_OUT[0] unconnected pin in architecture.
Warning 27: WE_MUX[0].CE[0] unconnected pin in architecture.
Warning 28: WE_MUX[0].WE[0] unconnected pin in architecture.
Warning 29: WE_MUX[0].WE_OUT[0] unconnected pin in architecture.
Warning 30: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 31: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 32: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 33: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 34: CEUSEDMUX[0].CE[0] unconnected pin in architecture.
Warning 35: SRUSEDMUX[0].SR[0] unconnected pin in architecture.
Warning 36: WE_MUX[0].CE[0] unconnected pin in architecture.
Warning 37: WE_MUX[0].WE[0] unconnected pin in architecture.
Warning 38: WE_MUX[0].WE_OUT[0] unconnected pin in architecture.
Warning 39: WE_MUX[0].CE[0] unconnected pin in architecture.
Warning 40: WE_MUX[0].WE[0] unconnected pin in architecture.
Warning 41: WE_MUX[0].WE_OUT[0] unconnected pin in architecture.
Warning 42: BRAM_L[0].BRAM_FIFO36_INJECTDBITERR[0] unconnected pin in architecture.
Warning 43: BRAM_L[0].BRAM_FIFO36_INJECTSBITERR[0] unconnected pin in architecture.
Warning 44: BRAM_L[0].BRAM_FIFO36_TSTBRAMRST[0] unconnected pin in architecture.
Warning 45: BRAM_L[0].BRAM_FIFO36_TSTCNT0[0] unconnected pin in architecture.
Warning 46: BRAM_L[0].BRAM_FIFO36_TSTCNT1[0] unconnected pin in architecture.
Warning 47: BRAM_L[0].BRAM_FIFO36_TSTCNT10[0] unconnected pin in architecture.
Warning 48: BRAM_L[0].BRAM_FIFO36_TSTCNT11[0] unconnected pin in architecture.
Warning 49: BRAM_L[0].BRAM_FIFO36_TSTCNT12[0] unconnected pin in architecture.
Warning 50: BRAM_L[0].BRAM_FIFO36_TSTCNT2[0] unconnected pin in architecture.
Warning 51: BRAM_L[0].BRAM_FIFO36_TSTCNT3[0] unconnected pin in architecture.
Warning 52: BRAM_L[0].BRAM_FIFO36_TSTCNT4[0] unconnected pin in architecture.
Warning 53: BRAM_L[0].BRAM_FIFO36_TSTCNT5[0] unconnected pin in architecture.
Warning 54: BRAM_L[0].BRAM_FIFO36_TSTCNT6[0] unconnected pin in architecture.
Warning 55: BRAM_L[0].BRAM_FIFO36_TSTCNT7[0] unconnected pin in architecture.
Warning 56: BRAM_L[0].BRAM_FIFO36_TSTCNT8[0] unconnected pin in architecture.
Warning 57: BRAM_L[0].BRAM_FIFO36_TSTCNT9[0] unconnected pin in architecture.
Warning 58: BRAM_L[0].BRAM_FIFO36_TSTFLAGIN[0] unconnected pin in architecture.
Warning 59: BRAM_L[0].BRAM_FIFO36_TSTIN0[0] unconnected pin in architecture.
Warning 60: BRAM_L[0].BRAM_FIFO36_TSTIN1[0] unconnected pin in architecture.
Warning 61: BRAM_L[0].BRAM_FIFO36_TSTIN2[0] unconnected pin in architecture.
Warning 62: BRAM_L[0].BRAM_FIFO36_TSTIN3[0] unconnected pin in architecture.
Warning 63: BRAM_L[0].BRAM_FIFO36_TSTIN4[0] unconnected pin in architecture.
Warning 64: BRAM_L[0].BRAM_FIFO36_TSTOFF[0] unconnected pin in architecture.
Warning 65: BRAM_L[0].BRAM_FIFO36_TSTRDCNTOFF[0] unconnected pin in architecture.
Warning 66: BRAM_L[0].BRAM_FIFO36_TSTRDOS0[0] unconnected pin in architecture.
Warning 67: BRAM_L[0].BRAM_FIFO36_TSTRDOS1[0] unconnected pin in architecture.
Warning 68: BRAM_L[0].BRAM_FIFO36_TSTRDOS10[0] unconnected pin in architecture.
Warning 69: BRAM_L[0].BRAM_FIFO36_TSTRDOS11[0] unconnected pin in architecture.
Warning 70: BRAM_L[0].BRAM_FIFO36_TSTRDOS12[0] unconnected pin in architecture.
Warning 71: BRAM_L[0].BRAM_FIFO36_TSTRDOS2[0] unconnected pin in architecture.
Warning 72: BRAM_L[0].BRAM_FIFO36_TSTRDOS3[0] unconnected pin in architecture.
Warning 73: BRAM_L[0].BRAM_FIFO36_TSTRDOS4[0] unconnected pin in architecture.
Warning 74: BRAM_L[0].BRAM_FIFO36_TSTRDOS5[0] unconnected pin in architecture.
Warning 75: BRAM_L[0].BRAM_FIFO36_TSTRDOS6[0] unconnected pin in architecture.
Warning 76: BRAM_L[0].BRAM_FIFO36_TSTRDOS7[0] unconnected pin in architecture.
Warning 77: BRAM_L[0].BRAM_FIFO36_TSTRDOS8[0] unconnected pin in architecture.
Warning 78: BRAM_L[0].BRAM_FIFO36_TSTRDOS9[0] unconnected pin in architecture.
Warning 79: BRAM_L[0].BRAM_FIFO36_TSTWRCNTOFF[0] unconnected pin in architecture.
Warning 80: BRAM_L[0].BRAM_FIFO36_TSTWROS0[0] unconnected pin in architecture.
Warning 81: BRAM_L[0].BRAM_FIFO36_TSTWROS1[0] unconnected pin in architecture.
Warning 82: BRAM_L[0].BRAM_FIFO36_TSTWROS10[0] unconnected pin in architecture.
Warning 83: BRAM_L[0].BRAM_FIFO36_TSTWROS11[0] unconnected pin in architecture.
Warning 84: BRAM_L[0].BRAM_FIFO36_TSTWROS12[0] unconnected pin in architecture.
Warning 85: BRAM_L[0].BRAM_FIFO36_TSTWROS2[0] unconnected pin in architecture.
Warning 86: BRAM_L[0].BRAM_FIFO36_TSTWROS3[0] unconnected pin in architecture.
Warning 87: BRAM_L[0].BRAM_FIFO36_TSTWROS4[0] unconnected pin in architecture.
Warning 88: BRAM_L[0].BRAM_FIFO36_TSTWROS5[0] unconnected pin in architecture.
Warning 89: BRAM_L[0].BRAM_FIFO36_TSTWROS6[0] unconnected pin in architecture.
Warning 90: BRAM_L[0].BRAM_FIFO36_TSTWROS7[0] unconnected pin in architecture.
Warning 91: BRAM_L[0].BRAM_FIFO36_TSTWROS8[0] unconnected pin in architecture.
Warning 92: BRAM_L[0].BRAM_FIFO36_TSTWROS9[0] unconnected pin in architecture.
Warning 93: BRAM_L[0].BRAM_FIFO18_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 94: BRAM_L[0].BRAM_FIFO18_ALMOSTFULL[0] unconnected pin in architecture.
Warning 95: BRAM_L[0].BRAM_FIFO18_EMPTY[0] unconnected pin in architecture.
Warning 96: BRAM_L[0].BRAM_FIFO18_FULL[0] unconnected pin in architecture.
Warning 97: BRAM_L[0].BRAM_FIFO18_RDCOUNT0[0] unconnected pin in architecture.
Warning 98: BRAM_L[0].BRAM_FIFO18_RDCOUNT1[0] unconnected pin in architecture.
Warning 99: BRAM_L[0].BRAM_FIFO18_RDCOUNT10[0] unconnected pin in architecture.
Warning 100: BRAM_L[0].BRAM_FIFO18_RDCOUNT11[0] unconnected pin in architecture.
Warning 101: BRAM_L[0].BRAM_FIFO18_RDCOUNT2[0] unconnected pin in architecture.
Warning 102: BRAM_L[0].BRAM_FIFO18_RDCOUNT3[0] unconnected pin in architecture.
Warning 103: BRAM_L[0].BRAM_FIFO18_RDCOUNT4[0] unconnected pin in architecture.
Warning 104: BRAM_L[0].BRAM_FIFO18_RDCOUNT5[0] unconnected pin in architecture.
Warning 105: BRAM_L[0].BRAM_FIFO18_RDCOUNT6[0] unconnected pin in architecture.
Warning 106: BRAM_L[0].BRAM_FIFO18_RDCOUNT7[0] unconnected pin in architecture.
Warning 107: BRAM_L[0].BRAM_FIFO18_RDCOUNT8[0] unconnected pin in architecture.
Warning 108: BRAM_L[0].BRAM_FIFO18_RDCOUNT9[0] unconnected pin in architecture.
Warning 109: BRAM_L[0].BRAM_FIFO18_RDERR[0] unconnected pin in architecture.
Warning 110: BRAM_L[0].BRAM_FIFO18_WRCOUNT0[0] unconnected pin in architecture.
Warning 111: BRAM_L[0].BRAM_FIFO18_WRCOUNT1[0] unconnected pin in architecture.
Warning 112: BRAM_L[0].BRAM_FIFO18_WRCOUNT10[0] unconnected pin in architecture.
Warning 113: BRAM_L[0].BRAM_FIFO18_WRCOUNT11[0] unconnected pin in architecture.
Warning 114: BRAM_L[0].BRAM_FIFO18_WRCOUNT2[0] unconnected pin in architecture.
Warning 115: BRAM_L[0].BRAM_FIFO18_WRCOUNT3[0] unconnected pin in architecture.
Warning 116: BRAM_L[0].BRAM_FIFO18_WRCOUNT4[0] unconnected pin in architecture.
Warning 117: BRAM_L[0].BRAM_FIFO18_WRCOUNT5[0] unconnected pin in architecture.
Warning 118: BRAM_L[0].BRAM_FIFO18_WRCOUNT6[0] unconnected pin in architecture.
Warning 119: BRAM_L[0].BRAM_FIFO18_WRCOUNT7[0] unconnected pin in architecture.
Warning 120: BRAM_L[0].BRAM_FIFO18_WRCOUNT8[0] unconnected pin in architecture.
Warning 121: BRAM_L[0].BRAM_FIFO18_WRCOUNT9[0] unconnected pin in architecture.
Warning 122: BRAM_L[0].BRAM_FIFO18_WRERR[0] unconnected pin in architecture.
Warning 123: BRAM_L[0].BRAM_FIFO36_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 124: BRAM_L[0].BRAM_FIFO36_ALMOSTFULL[0] unconnected pin in architecture.
Warning 125: BRAM_L[0].BRAM_FIFO36_DBITERR[0] unconnected pin in architecture.
Warning 126: BRAM_L[0].BRAM_FIFO36_ECCPARITY0[0] unconnected pin in architecture.
Warning 127: BRAM_L[0].BRAM_FIFO36_ECCPARITY1[0] unconnected pin in architecture.
Warning 128: BRAM_L[0].BRAM_FIFO36_ECCPARITY2[0] unconnected pin in architecture.
Warning 129: BRAM_L[0].BRAM_FIFO36_ECCPARITY3[0] unconnected pin in architecture.
Warning 130: BRAM_L[0].BRAM_FIFO36_ECCPARITY4[0] unconnected pin in architecture.
Warning 131: BRAM_L[0].BRAM_FIFO36_ECCPARITY5[0] unconnected pin in architecture.
Warning 132: BRAM_L[0].BRAM_FIFO36_ECCPARITY6[0] unconnected pin in architecture.
Warning 133: BRAM_L[0].BRAM_FIFO36_ECCPARITY7[0] unconnected pin in architecture.
Warning 134: BRAM_L[0].BRAM_FIFO36_EMPTY[0] unconnected pin in architecture.
Warning 135: BRAM_L[0].BRAM_FIFO36_FULL[0] unconnected pin in architecture.
Warning 136: BRAM_L[0].BRAM_FIFO36_RDCOUNT0[0] unconnected pin in architecture.
Warning 137: BRAM_L[0].BRAM_FIFO36_RDCOUNT1[0] unconnected pin in architecture.
Warning 138: BRAM_L[0].BRAM_FIFO36_RDCOUNT10[0] unconnected pin in architecture.
Warning 139: BRAM_L[0].BRAM_FIFO36_RDCOUNT11[0] unconnected pin in architecture.
Warning 140: BRAM_L[0].BRAM_FIFO36_RDCOUNT12[0] unconnected pin in architecture.
Warning 141: BRAM_L[0].BRAM_FIFO36_RDCOUNT2[0] unconnected pin in architecture.
Warning 142: BRAM_L[0].BRAM_FIFO36_RDCOUNT3[0] unconnected pin in architecture.
Warning 143: BRAM_L[0].BRAM_FIFO36_RDCOUNT4[0] unconnected pin in architecture.
Warning 144: BRAM_L[0].BRAM_FIFO36_RDCOUNT5[0] unconnected pin in architecture.
Warning 145: BRAM_L[0].BRAM_FIFO36_RDCOUNT6[0] unconnected pin in architecture.
Warning 146: BRAM_L[0].BRAM_FIFO36_RDCOUNT7[0] unconnected pin in architecture.
Warning 147: BRAM_L[0].BRAM_FIFO36_RDCOUNT8[0] unconnected pin in architecture.
Warning 148: BRAM_L[0].BRAM_FIFO36_RDCOUNT9[0] unconnected pin in architecture.
Warning 149: BRAM_L[0].BRAM_FIFO36_RDERR[0] unconnected pin in architecture.
Warning 150: BRAM_L[0].BRAM_FIFO36_SBITERR[0] unconnected pin in architecture.
Warning 151: BRAM_L[0].BRAM_FIFO36_TSTOUT0[0] unconnected pin in architecture.
Warning 152: BRAM_L[0].BRAM_FIFO36_TSTOUT1[0] unconnected pin in architecture.
Warning 153: BRAM_L[0].BRAM_FIFO36_TSTOUT2[0] unconnected pin in architecture.
Warning 154: BRAM_L[0].BRAM_FIFO36_TSTOUT3[0] unconnected pin in architecture.
Warning 155: BRAM_L[0].BRAM_FIFO36_TSTOUT4[0] unconnected pin in architecture.
Warning 156: BRAM_L[0].BRAM_FIFO36_WRCOUNT0[0] unconnected pin in architecture.
Warning 157: BRAM_L[0].BRAM_FIFO36_WRCOUNT1[0] unconnected pin in architecture.
Warning 158: BRAM_L[0].BRAM_FIFO36_WRCOUNT10[0] unconnected pin in architecture.
Warning 159: BRAM_L[0].BRAM_FIFO36_WRCOUNT11[0] unconnected pin in architecture.
Warning 160: BRAM_L[0].BRAM_FIFO36_WRCOUNT12[0] unconnected pin in architecture.
Warning 161: BRAM_L[0].BRAM_FIFO36_WRCOUNT2[0] unconnected pin in architecture.
Warning 162: BRAM_L[0].BRAM_FIFO36_WRCOUNT3[0] unconnected pin in architecture.
Warning 163: BRAM_L[0].BRAM_FIFO36_WRCOUNT4[0] unconnected pin in architecture.
Warning 164: BRAM_L[0].BRAM_FIFO36_WRCOUNT5[0] unconnected pin in architecture.
Warning 165: BRAM_L[0].BRAM_FIFO36_WRCOUNT6[0] unconnected pin in architecture.
Warning 166: BRAM_L[0].BRAM_FIFO36_WRCOUNT7[0] unconnected pin in architecture.
Warning 167: BRAM_L[0].BRAM_FIFO36_WRCOUNT8[0] unconnected pin in architecture.
Warning 168: BRAM_L[0].BRAM_FIFO36_WRCOUNT9[0] unconnected pin in architecture.
Warning 169: BRAM_L[0].BRAM_FIFO36_WRERR[0] unconnected pin in architecture.
Warning 170: BRAM_L[0].BRAM_RAMB18_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 171: BRAM_L[0].BRAM_RAMB18_ALMOSTFULL[0] unconnected pin in architecture.
Warning 172: BRAM_L[0].BRAM_RAMB18_EMPTY[0] unconnected pin in architecture.
Warning 173: BRAM_L[0].BRAM_RAMB18_FULL[0] unconnected pin in architecture.
Warning 174: BRAM_L[0].BRAM_RAMB18_RDCOUNT0[0] unconnected pin in architecture.
Warning 175: BRAM_L[0].BRAM_RAMB18_RDCOUNT1[0] unconnected pin in architecture.
Warning 176: BRAM_L[0].BRAM_RAMB18_RDCOUNT10[0] unconnected pin in architecture.
Warning 177: BRAM_L[0].BRAM_RAMB18_RDCOUNT11[0] unconnected pin in architecture.
Warning 178: BRAM_L[0].BRAM_RAMB18_RDCOUNT2[0] unconnected pin in architecture.
Warning 179: BRAM_L[0].BRAM_RAMB18_RDCOUNT3[0] unconnected pin in architecture.
Warning 180: BRAM_L[0].BRAM_RAMB18_RDCOUNT4[0] unconnected pin in architecture.
Warning 181: BRAM_L[0].BRAM_RAMB18_RDCOUNT5[0] unconnected pin in architecture.
Warning 182: BRAM_L[0].BRAM_RAMB18_RDCOUNT6[0] unconnected pin in architecture.
Warning 183: BRAM_L[0].BRAM_RAMB18_RDCOUNT7[0] unconnected pin in architecture.
Warning 184: BRAM_L[0].BRAM_RAMB18_RDCOUNT8[0] unconnected pin in architecture.
Warning 185: BRAM_L[0].BRAM_RAMB18_RDCOUNT9[0] unconnected pin in architecture.
Warning 186: BRAM_L[0].BRAM_RAMB18_RDERR[0] unconnected pin in architecture.
Warning 187: BRAM_L[0].BRAM_RAMB18_WRCOUNT0[0] unconnected pin in architecture.
Warning 188: BRAM_L[0].BRAM_RAMB18_WRCOUNT1[0] unconnected pin in architecture.
Warning 189: BRAM_L[0].BRAM_RAMB18_WRCOUNT10[0] unconnected pin in architecture.
Warning 190: BRAM_L[0].BRAM_RAMB18_WRCOUNT11[0] unconnected pin in architecture.
Warning 191: BRAM_L[0].BRAM_RAMB18_WRCOUNT2[0] unconnected pin in architecture.
Warning 192: BRAM_L[0].BRAM_RAMB18_WRCOUNT3[0] unconnected pin in architecture.
Warning 193: BRAM_L[0].BRAM_RAMB18_WRCOUNT4[0] unconnected pin in architecture.
Warning 194: BRAM_L[0].BRAM_RAMB18_WRCOUNT5[0] unconnected pin in architecture.
Warning 195: BRAM_L[0].BRAM_RAMB18_WRCOUNT6[0] unconnected pin in architecture.
Warning 196: BRAM_L[0].BRAM_RAMB18_WRCOUNT7[0] unconnected pin in architecture.
Warning 197: BRAM_L[0].BRAM_RAMB18_WRCOUNT8[0] unconnected pin in architecture.
Warning 198: BRAM_L[0].BRAM_RAMB18_WRCOUNT9[0] unconnected pin in architecture.
Warning 199: BRAM_L[0].BRAM_RAMB18_WRERR[0] unconnected pin in architecture.
Warning 200: BRAM_L[0].BRAM_FIFO36_REGCLKARDRCLKL[0] unconnected pin in architecture.
Warning 201: BRAM_L[0].BRAM_FIFO36_REGCLKARDRCLKU[0] unconnected pin in architecture.
Warning 202: BRAM_L[0].BRAM_FIFO36_REGCLKBL[0] unconnected pin in architecture.
Warning 203: BRAM_L[0].BRAM_FIFO36_REGCLKBU[0] unconnected pin in architecture.
Warning 204: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[0] unconnected pin in architecture.
Warning 205: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[1] unconnected pin in architecture.
Warning 206: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[2] unconnected pin in architecture.
Warning 207: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[3] unconnected pin in architecture.
Warning 208: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[4] unconnected pin in architecture.
Warning 209: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[5] unconnected pin in architecture.
Warning 210: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[6] unconnected pin in architecture.
Warning 211: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[7] unconnected pin in architecture.
Warning 212: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[8] unconnected pin in architecture.
Warning 213: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[9] unconnected pin in architecture.
Warning 214: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[10] unconnected pin in architecture.
Warning 215: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[11] unconnected pin in architecture.
Warning 216: BRAM_X0[0].RAMB36_X0Y0_TSTWROS[12] unconnected pin in architecture.
Warning 217: BRAM_X0[0].RAMB36_X0Y0_TSTWRCNTOFF[0] unconnected pin in architecture.
Warning 218: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[0] unconnected pin in architecture.
Warning 219: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[1] unconnected pin in architecture.
Warning 220: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[2] unconnected pin in architecture.
Warning 221: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[3] unconnected pin in architecture.
Warning 222: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[4] unconnected pin in architecture.
Warning 223: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[5] unconnected pin in architecture.
Warning 224: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[6] unconnected pin in architecture.
Warning 225: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[7] unconnected pin in architecture.
Warning 226: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[8] unconnected pin in architecture.
Warning 227: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[9] unconnected pin in architecture.
Warning 228: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[10] unconnected pin in architecture.
Warning 229: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[11] unconnected pin in architecture.
Warning 230: BRAM_X0[0].RAMB36_X0Y0_TSTRDOS[12] unconnected pin in architecture.
Warning 231: BRAM_X0[0].RAMB36_X0Y0_TSTRDCNTOFF[0] unconnected pin in architecture.
Warning 232: BRAM_X0[0].RAMB36_X0Y0_TSTOFF[0] unconnected pin in architecture.
Warning 233: BRAM_X0[0].RAMB36_X0Y0_TSTIN[0] unconnected pin in architecture.
Warning 234: BRAM_X0[0].RAMB36_X0Y0_TSTIN[1] unconnected pin in architecture.
Warning 235: BRAM_X0[0].RAMB36_X0Y0_TSTIN[2] unconnected pin in architecture.
Warning 236: BRAM_X0[0].RAMB36_X0Y0_TSTIN[3] unconnected pin in architecture.
Warning 237: BRAM_X0[0].RAMB36_X0Y0_TSTIN[4] unconnected pin in architecture.
Warning 238: BRAM_X0[0].RAMB36_X0Y0_TSTFLAGIN[0] unconnected pin in architecture.
Warning 239: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[0] unconnected pin in architecture.
Warning 240: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[1] unconnected pin in architecture.
Warning 241: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[2] unconnected pin in architecture.
Warning 242: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[3] unconnected pin in architecture.
Warning 243: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[4] unconnected pin in architecture.
Warning 244: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[5] unconnected pin in architecture.
Warning 245: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[6] unconnected pin in architecture.
Warning 246: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[7] unconnected pin in architecture.
Warning 247: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[8] unconnected pin in architecture.
Warning 248: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[9] unconnected pin in architecture.
Warning 249: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[10] unconnected pin in architecture.
Warning 250: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[11] unconnected pin in architecture.
Warning 251: BRAM_X0[0].RAMB36_X0Y0_TSTCNT[12] unconnected pin in architecture.
Warning 252: BRAM_X0[0].RAMB36_X0Y0_TSTBRAMRST[0] unconnected pin in architecture.
Warning 253: BRAM_X0[0].RAMB36_X0Y0_REGCLKBU[0] unconnected pin in architecture.
Warning 254: BRAM_X0[0].RAMB36_X0Y0_REGCLKBL[0] unconnected pin in architecture.
Warning 255: BRAM_X0[0].RAMB36_X0Y0_REGCLKARDRCLKU[0] unconnected pin in architecture.
Warning 256: BRAM_X0[0].RAMB36_X0Y0_REGCLKARDRCLKL[0] unconnected pin in architecture.
Warning 257: BRAM_X0[0].RAMB36_X0Y0_INJECTDBITERR[0] unconnected pin in architecture.
Warning 258: BRAM_X0[0].RAMB36_X0Y0_INJECTSBITERR[0] unconnected pin in architecture.
Warning 259: BRAM_X0[0].RAMB36_X0Y0_WRERR[0] unconnected pin in architecture.
Warning 260: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[0] unconnected pin in architecture.
Warning 261: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[1] unconnected pin in architecture.
Warning 262: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[2] unconnected pin in architecture.
Warning 263: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[3] unconnected pin in architecture.
Warning 264: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[4] unconnected pin in architecture.
Warning 265: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[5] unconnected pin in architecture.
Warning 266: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[6] unconnected pin in architecture.
Warning 267: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[7] unconnected pin in architecture.
Warning 268: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[8] unconnected pin in architecture.
Warning 269: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[9] unconnected pin in architecture.
Warning 270: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[10] unconnected pin in architecture.
Warning 271: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[11] unconnected pin in architecture.
Warning 272: BRAM_X0[0].RAMB36_X0Y0_WRCOUNT[12] unconnected pin in architecture.
Warning 273: BRAM_X0[0].RAMB36_X0Y0_TSTOUT[0] unconnected pin in architecture.
Warning 274: BRAM_X0[0].RAMB36_X0Y0_TSTOUT[1] unconnected pin in architecture.
Warning 275: BRAM_X0[0].RAMB36_X0Y0_TSTOUT[2] unconnected pin in architecture.
Warning 276: BRAM_X0[0].RAMB36_X0Y0_TSTOUT[3] unconnected pin in architecture.
Warning 277: BRAM_X0[0].RAMB36_X0Y0_TSTOUT[4] unconnected pin in architecture.
Warning 278: BRAM_X0[0].RAMB36_X0Y0_SBITERR[0] unconnected pin in architecture.
Warning 279: BRAM_X0[0].RAMB36_X0Y0_RDERR[0] unconnected pin in architecture.
Warning 280: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[0] unconnected pin in architecture.
Warning 281: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[1] unconnected pin in architecture.
Warning 282: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[2] unconnected pin in architecture.
Warning 283: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[3] unconnected pin in architecture.
Warning 284: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[4] unconnected pin in architecture.
Warning 285: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[5] unconnected pin in architecture.
Warning 286: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[6] unconnected pin in architecture.
Warning 287: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[7] unconnected pin in architecture.
Warning 288: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[8] unconnected pin in architecture.
Warning 289: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[9] unconnected pin in architecture.
Warning 290: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[10] unconnected pin in architecture.
Warning 291: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[11] unconnected pin in architecture.
Warning 292: BRAM_X0[0].RAMB36_X0Y0_RDCOUNT[12] unconnected pin in architecture.
Warning 293: BRAM_X0[0].RAMB36_X0Y0_FULL[0] unconnected pin in architecture.
Warning 294: BRAM_X0[0].RAMB36_X0Y0_EMPTY[0] unconnected pin in architecture.
Warning 295: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[0] unconnected pin in architecture.
Warning 296: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[1] unconnected pin in architecture.
Warning 297: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[2] unconnected pin in architecture.
Warning 298: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[3] unconnected pin in architecture.
Warning 299: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[4] unconnected pin in architecture.
Warning 300: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[5] unconnected pin in architecture.
Warning 301: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[6] unconnected pin in architecture.
Warning 302: BRAM_X0[0].RAMB36_X0Y0_ECCPARITY[7] unconnected pin in architecture.
Warning 303: BRAM_X0[0].RAMB36_X0Y0_DBITERR[0] unconnected pin in architecture.
Warning 304: BRAM_X0[0].RAMB36_X0Y0_ALMOSTFULL[0] unconnected pin in architecture.
Warning 305: BRAM_X0[0].RAMB36_X0Y0_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 306: BRAM_X0[0].RAMB18_X0Y0_WRERR[0] unconnected pin in architecture.
Warning 307: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[0] unconnected pin in architecture.
Warning 308: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[1] unconnected pin in architecture.
Warning 309: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[2] unconnected pin in architecture.
Warning 310: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[3] unconnected pin in architecture.
Warning 311: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[4] unconnected pin in architecture.
Warning 312: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[5] unconnected pin in architecture.
Warning 313: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[6] unconnected pin in architecture.
Warning 314: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[7] unconnected pin in architecture.
Warning 315: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[8] unconnected pin in architecture.
Warning 316: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[9] unconnected pin in architecture.
Warning 317: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[10] unconnected pin in architecture.
Warning 318: BRAM_X0[0].RAMB18_X0Y0_WRCOUNT[11] unconnected pin in architecture.
Warning 319: BRAM_X0[0].RAMB18_X0Y0_RDERR[0] unconnected pin in architecture.
Warning 320: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[0] unconnected pin in architecture.
Warning 321: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[1] unconnected pin in architecture.
Warning 322: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[2] unconnected pin in architecture.
Warning 323: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[3] unconnected pin in architecture.
Warning 324: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[4] unconnected pin in architecture.
Warning 325: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[5] unconnected pin in architecture.
Warning 326: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[6] unconnected pin in architecture.
Warning 327: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[7] unconnected pin in architecture.
Warning 328: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[8] unconnected pin in architecture.
Warning 329: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[9] unconnected pin in architecture.
Warning 330: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[10] unconnected pin in architecture.
Warning 331: BRAM_X0[0].RAMB18_X0Y0_RDCOUNT[11] unconnected pin in architecture.
Warning 332: BRAM_X0[0].RAMB18_X0Y0_FULL[0] unconnected pin in architecture.
Warning 333: BRAM_X0[0].RAMB18_X0Y0_EMPTY[0] unconnected pin in architecture.
Warning 334: BRAM_X0[0].RAMB18_X0Y0_ALMOSTFULL[0] unconnected pin in architecture.
Warning 335: BRAM_X0[0].RAMB18_X0Y0_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 336: BRAM_X0[0].RAMB18_X0Y1_WRERR[0] unconnected pin in architecture.
Warning 337: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[0] unconnected pin in architecture.
Warning 338: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[1] unconnected pin in architecture.
Warning 339: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[2] unconnected pin in architecture.
Warning 340: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[3] unconnected pin in architecture.
Warning 341: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[4] unconnected pin in architecture.
Warning 342: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[5] unconnected pin in architecture.
Warning 343: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[6] unconnected pin in architecture.
Warning 344: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[7] unconnected pin in architecture.
Warning 345: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[8] unconnected pin in architecture.
Warning 346: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[9] unconnected pin in architecture.
Warning 347: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[10] unconnected pin in architecture.
Warning 348: BRAM_X0[0].RAMB18_X0Y1_WRCOUNT[11] unconnected pin in architecture.
Warning 349: BRAM_X0[0].RAMB18_X0Y1_RDERR[0] unconnected pin in architecture.
Warning 350: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[0] unconnected pin in architecture.
Warning 351: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[1] unconnected pin in architecture.
Warning 352: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[2] unconnected pin in architecture.
Warning 353: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[3] unconnected pin in architecture.
Warning 354: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[4] unconnected pin in architecture.
Warning 355: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[5] unconnected pin in architecture.
Warning 356: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[6] unconnected pin in architecture.
Warning 357: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[7] unconnected pin in architecture.
Warning 358: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[8] unconnected pin in architecture.
Warning 359: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[9] unconnected pin in architecture.
Warning 360: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[10] unconnected pin in architecture.
Warning 361: BRAM_X0[0].RAMB18_X0Y1_RDCOUNT[11] unconnected pin in architecture.
Warning 362: BRAM_X0[0].RAMB18_X0Y1_FULL[0] unconnected pin in architecture.
Warning 363: BRAM_X0[0].RAMB18_X0Y1_EMPTY[0] unconnected pin in architecture.
Warning 364: BRAM_X0[0].RAMB18_X0Y1_ALMOSTFULL[0] unconnected pin in architecture.
Warning 365: BRAM_X0[0].RAMB18_X0Y1_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 366: BRAM[0].RAMB36_Y0_TSTWROS[0] unconnected pin in architecture.
Warning 367: BRAM[0].RAMB36_Y0_TSTWROS[1] unconnected pin in architecture.
Warning 368: BRAM[0].RAMB36_Y0_TSTWROS[2] unconnected pin in architecture.
Warning 369: BRAM[0].RAMB36_Y0_TSTWROS[3] unconnected pin in architecture.
Warning 370: BRAM[0].RAMB36_Y0_TSTWROS[4] unconnected pin in architecture.
Warning 371: BRAM[0].RAMB36_Y0_TSTWROS[5] unconnected pin in architecture.
Warning 372: BRAM[0].RAMB36_Y0_TSTWROS[6] unconnected pin in architecture.
Warning 373: BRAM[0].RAMB36_Y0_TSTWROS[7] unconnected pin in architecture.
Warning 374: BRAM[0].RAMB36_Y0_TSTWROS[8] unconnected pin in architecture.
Warning 375: BRAM[0].RAMB36_Y0_TSTWROS[9] unconnected pin in architecture.
Warning 376: BRAM[0].RAMB36_Y0_TSTWROS[10] unconnected pin in architecture.
Warning 377: BRAM[0].RAMB36_Y0_TSTWROS[11] unconnected pin in architecture.
Warning 378: BRAM[0].RAMB36_Y0_TSTWROS[12] unconnected pin in architecture.
Warning 379: BRAM[0].RAMB36_Y0_TSTWRCNTOFF[0] unconnected pin in architecture.
Warning 380: BRAM[0].RAMB36_Y0_TSTRDOS[0] unconnected pin in architecture.
Warning 381: BRAM[0].RAMB36_Y0_TSTRDOS[1] unconnected pin in architecture.
Warning 382: BRAM[0].RAMB36_Y0_TSTRDOS[2] unconnected pin in architecture.
Warning 383: BRAM[0].RAMB36_Y0_TSTRDOS[3] unconnected pin in architecture.
Warning 384: BRAM[0].RAMB36_Y0_TSTRDOS[4] unconnected pin in architecture.
Warning 385: BRAM[0].RAMB36_Y0_TSTRDOS[5] unconnected pin in architecture.
Warning 386: BRAM[0].RAMB36_Y0_TSTRDOS[6] unconnected pin in architecture.
Warning 387: BRAM[0].RAMB36_Y0_TSTRDOS[7] unconnected pin in architecture.
Warning 388: BRAM[0].RAMB36_Y0_TSTRDOS[8] unconnected pin in architecture.
Warning 389: BRAM[0].RAMB36_Y0_TSTRDOS[9] unconnected pin in architecture.
Warning 390: BRAM[0].RAMB36_Y0_TSTRDOS[10] unconnected pin in architecture.
Warning 391: BRAM[0].RAMB36_Y0_TSTRDOS[11] unconnected pin in architecture.
Warning 392: BRAM[0].RAMB36_Y0_TSTRDOS[12] unconnected pin in architecture.
Warning 393: BRAM[0].RAMB36_Y0_TSTRDCNTOFF[0] unconnected pin in architecture.
Warning 394: BRAM[0].RAMB36_Y0_TSTOFF[0] unconnected pin in architecture.
Warning 395: BRAM[0].RAMB36_Y0_TSTIN[0] unconnected pin in architecture.
Warning 396: BRAM[0].RAMB36_Y0_TSTIN[1] unconnected pin in architecture.
Warning 397: BRAM[0].RAMB36_Y0_TSTIN[2] unconnected pin in architecture.
Warning 398: BRAM[0].RAMB36_Y0_TSTIN[3] unconnected pin in architecture.
Warning 399: BRAM[0].RAMB36_Y0_TSTIN[4] unconnected pin in architecture.
Warning 400: BRAM[0].RAMB36_Y0_TSTFLAGIN[0] unconnected pin in architecture.
Warning 401: BRAM[0].RAMB36_Y0_TSTCNT[0] unconnected pin in architecture.
Warning 402: BRAM[0].RAMB36_Y0_TSTCNT[1] unconnected pin in architecture.
Warning 403: BRAM[0].RAMB36_Y0_TSTCNT[2] unconnected pin in architecture.
Warning 404: BRAM[0].RAMB36_Y0_TSTCNT[3] unconnected pin in architecture.
Warning 405: BRAM[0].RAMB36_Y0_TSTCNT[4] unconnected pin in architecture.
Warning 406: BRAM[0].RAMB36_Y0_TSTCNT[5] unconnected pin in architecture.
Warning 407: BRAM[0].RAMB36_Y0_TSTCNT[6] unconnected pin in architecture.
Warning 408: BRAM[0].RAMB36_Y0_TSTCNT[7] unconnected pin in architecture.
Warning 409: BRAM[0].RAMB36_Y0_TSTCNT[8] unconnected pin in architecture.
Warning 410: BRAM[0].RAMB36_Y0_TSTCNT[9] unconnected pin in architecture.
Warning 411: BRAM[0].RAMB36_Y0_TSTCNT[10] unconnected pin in architecture.
Warning 412: BRAM[0].RAMB36_Y0_TSTCNT[11] unconnected pin in architecture.
Warning 413: BRAM[0].RAMB36_Y0_TSTCNT[12] unconnected pin in architecture.
Warning 414: BRAM[0].RAMB36_Y0_TSTBRAMRST[0] unconnected pin in architecture.
Warning 415: BRAM[0].RAMB36_Y0_REGCLKBU[0] unconnected pin in architecture.
Warning 416: BRAM[0].RAMB36_Y0_REGCLKBL[0] unconnected pin in architecture.
Warning 417: BRAM[0].RAMB36_Y0_REGCLKARDRCLKU[0] unconnected pin in architecture.
Warning 418: BRAM[0].RAMB36_Y0_REGCLKARDRCLKL[0] unconnected pin in architecture.
Warning 419: BRAM[0].RAMB36_Y0_INJECTDBITERR[0] unconnected pin in architecture.
Warning 420: BRAM[0].RAMB36_Y0_INJECTSBITERR[0] unconnected pin in architecture.
Warning 421: BRAM[0].RAMB36_Y0_WRERR[0] unconnected pin in architecture.
Warning 422: BRAM[0].RAMB36_Y0_WRCOUNT[0] unconnected pin in architecture.
Warning 423: BRAM[0].RAMB36_Y0_WRCOUNT[1] unconnected pin in architecture.
Warning 424: BRAM[0].RAMB36_Y0_WRCOUNT[2] unconnected pin in architecture.
Warning 425: BRAM[0].RAMB36_Y0_WRCOUNT[3] unconnected pin in architecture.
Warning 426: BRAM[0].RAMB36_Y0_WRCOUNT[4] unconnected pin in architecture.
Warning 427: BRAM[0].RAMB36_Y0_WRCOUNT[5] unconnected pin in architecture.
Warning 428: BRAM[0].RAMB36_Y0_WRCOUNT[6] unconnected pin in architecture.
Warning 429: BRAM[0].RAMB36_Y0_WRCOUNT[7] unconnected pin in architecture.
Warning 430: BRAM[0].RAMB36_Y0_WRCOUNT[8] unconnected pin in architecture.
Warning 431: BRAM[0].RAMB36_Y0_WRCOUNT[9] unconnected pin in architecture.
Warning 432: BRAM[0].RAMB36_Y0_WRCOUNT[10] unconnected pin in architecture.
Warning 433: BRAM[0].RAMB36_Y0_WRCOUNT[11] unconnected pin in architecture.
Warning 434: BRAM[0].RAMB36_Y0_WRCOUNT[12] unconnected pin in architecture.
Warning 435: BRAM[0].RAMB36_Y0_TSTOUT[0] unconnected pin in architecture.
Warning 436: BRAM[0].RAMB36_Y0_TSTOUT[1] unconnected pin in architecture.
Warning 437: BRAM[0].RAMB36_Y0_TSTOUT[2] unconnected pin in architecture.
Warning 438: BRAM[0].RAMB36_Y0_TSTOUT[3] unconnected pin in architecture.
Warning 439: BRAM[0].RAMB36_Y0_TSTOUT[4] unconnected pin in architecture.
Warning 440: BRAM[0].RAMB36_Y0_SBITERR[0] unconnected pin in architecture.
Warning 441: BRAM[0].RAMB36_Y0_RDERR[0] unconnected pin in architecture.
Warning 442: BRAM[0].RAMB36_Y0_RDCOUNT[0] unconnected pin in architecture.
Warning 443: BRAM[0].RAMB36_Y0_RDCOUNT[1] unconnected pin in architecture.
Warning 444: BRAM[0].RAMB36_Y0_RDCOUNT[2] unconnected pin in architecture.
Warning 445: BRAM[0].RAMB36_Y0_RDCOUNT[3] unconnected pin in architecture.
Warning 446: BRAM[0].RAMB36_Y0_RDCOUNT[4] unconnected pin in architecture.
Warning 447: BRAM[0].RAMB36_Y0_RDCOUNT[5] unconnected pin in architecture.
Warning 448: BRAM[0].RAMB36_Y0_RDCOUNT[6] unconnected pin in architecture.
Warning 449: BRAM[0].RAMB36_Y0_RDCOUNT[7] unconnected pin in architecture.
Warning 450: BRAM[0].RAMB36_Y0_RDCOUNT[8] unconnected pin in architecture.
Warning 451: BRAM[0].RAMB36_Y0_RDCOUNT[9] unconnected pin in architecture.
Warning 452: BRAM[0].RAMB36_Y0_RDCOUNT[10] unconnected pin in architecture.
Warning 453: BRAM[0].RAMB36_Y0_RDCOUNT[11] unconnected pin in architecture.
Warning 454: BRAM[0].RAMB36_Y0_RDCOUNT[12] unconnected pin in architecture.
Warning 455: BRAM[0].RAMB36_Y0_FULL[0] unconnected pin in architecture.
Warning 456: BRAM[0].RAMB36_Y0_EMPTY[0] unconnected pin in architecture.
Warning 457: BRAM[0].RAMB36_Y0_ECCPARITY[0] unconnected pin in architecture.
Warning 458: BRAM[0].RAMB36_Y0_ECCPARITY[1] unconnected pin in architecture.
Warning 459: BRAM[0].RAMB36_Y0_ECCPARITY[2] unconnected pin in architecture.
Warning 460: BRAM[0].RAMB36_Y0_ECCPARITY[3] unconnected pin in architecture.
Warning 461: BRAM[0].RAMB36_Y0_ECCPARITY[4] unconnected pin in architecture.
Warning 462: BRAM[0].RAMB36_Y0_ECCPARITY[5] unconnected pin in architecture.
Warning 463: BRAM[0].RAMB36_Y0_ECCPARITY[6] unconnected pin in architecture.
Warning 464: BRAM[0].RAMB36_Y0_ECCPARITY[7] unconnected pin in architecture.
Warning 465: BRAM[0].RAMB36_Y0_DBITERR[0] unconnected pin in architecture.
Warning 466: BRAM[0].RAMB36_Y0_ALMOSTFULL[0] unconnected pin in architecture.
Warning 467: BRAM[0].RAMB36_Y0_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 468: BRAM[0].RAMB18_Y0_WRERR[0] unconnected pin in architecture.
Warning 469: BRAM[0].RAMB18_Y0_WRCOUNT[0] unconnected pin in architecture.
Warning 470: BRAM[0].RAMB18_Y0_WRCOUNT[1] unconnected pin in architecture.
Warning 471: BRAM[0].RAMB18_Y0_WRCOUNT[2] unconnected pin in architecture.
Warning 472: BRAM[0].RAMB18_Y0_WRCOUNT[3] unconnected pin in architecture.
Warning 473: BRAM[0].RAMB18_Y0_WRCOUNT[4] unconnected pin in architecture.
Warning 474: BRAM[0].RAMB18_Y0_WRCOUNT[5] unconnected pin in architecture.
Warning 475: BRAM[0].RAMB18_Y0_WRCOUNT[6] unconnected pin in architecture.
Warning 476: BRAM[0].RAMB18_Y0_WRCOUNT[7] unconnected pin in architecture.
Warning 477: BRAM[0].RAMB18_Y0_WRCOUNT[8] unconnected pin in architecture.
Warning 478: BRAM[0].RAMB18_Y0_WRCOUNT[9] unconnected pin in architecture.
Warning 479: BRAM[0].RAMB18_Y0_WRCOUNT[10] unconnected pin in architecture.
Warning 480: BRAM[0].RAMB18_Y0_WRCOUNT[11] unconnected pin in architecture.
Warning 481: BRAM[0].RAMB18_Y0_RDERR[0] unconnected pin in architecture.
Warning 482: BRAM[0].RAMB18_Y0_RDCOUNT[0] unconnected pin in architecture.
Warning 483: BRAM[0].RAMB18_Y0_RDCOUNT[1] unconnected pin in architecture.
Warning 484: BRAM[0].RAMB18_Y0_RDCOUNT[2] unconnected pin in architecture.
Warning 485: BRAM[0].RAMB18_Y0_RDCOUNT[3] unconnected pin in architecture.
Warning 486: BRAM[0].RAMB18_Y0_RDCOUNT[4] unconnected pin in architecture.
Warning 487: BRAM[0].RAMB18_Y0_RDCOUNT[5] unconnected pin in architecture.
Warning 488: BRAM[0].RAMB18_Y0_RDCOUNT[6] unconnected pin in architecture.
Warning 489: BRAM[0].RAMB18_Y0_RDCOUNT[7] unconnected pin in architecture.
Warning 490: BRAM[0].RAMB18_Y0_RDCOUNT[8] unconnected pin in architecture.
Warning 491: BRAM[0].RAMB18_Y0_RDCOUNT[9] unconnected pin in architecture.
Warning 492: BRAM[0].RAMB18_Y0_RDCOUNT[10] unconnected pin in architecture.
Warning 493: BRAM[0].RAMB18_Y0_RDCOUNT[11] unconnected pin in architecture.
Warning 494: BRAM[0].RAMB18_Y0_FULL[0] unconnected pin in architecture.
Warning 495: BRAM[0].RAMB18_Y0_EMPTY[0] unconnected pin in architecture.
Warning 496: BRAM[0].RAMB18_Y0_ALMOSTFULL[0] unconnected pin in architecture.
Warning 497: BRAM[0].RAMB18_Y0_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 498: BRAM[0].RAMB18_Y1_WRERR[0] unconnected pin in architecture.
Warning 499: BRAM[0].RAMB18_Y1_WRCOUNT[0] unconnected pin in architecture.
Warning 500: BRAM[0].RAMB18_Y1_WRCOUNT[1] unconnected pin in architecture.
Warning 501: BRAM[0].RAMB18_Y1_WRCOUNT[2] unconnected pin in architecture.
Warning 502: BRAM[0].RAMB18_Y1_WRCOUNT[3] unconnected pin in architecture.
Warning 503: BRAM[0].RAMB18_Y1_WRCOUNT[4] unconnected pin in architecture.
Warning 504: BRAM[0].RAMB18_Y1_WRCOUNT[5] unconnected pin in architecture.
Warning 505: BRAM[0].RAMB18_Y1_WRCOUNT[6] unconnected pin in architecture.
Warning 506: BRAM[0].RAMB18_Y1_WRCOUNT[7] unconnected pin in architecture.
Warning 507: BRAM[0].RAMB18_Y1_WRCOUNT[8] unconnected pin in architecture.
Warning 508: BRAM[0].RAMB18_Y1_WRCOUNT[9] unconnected pin in architecture.
Warning 509: BRAM[0].RAMB18_Y1_WRCOUNT[10] unconnected pin in architecture.
Warning 510: BRAM[0].RAMB18_Y1_WRCOUNT[11] unconnected pin in architecture.
Warning 511: BRAM[0].RAMB18_Y1_RDERR[0] unconnected pin in architecture.
Warning 512: BRAM[0].RAMB18_Y1_RDCOUNT[0] unconnected pin in architecture.
Warning 513: BRAM[0].RAMB18_Y1_RDCOUNT[1] unconnected pin in architecture.
Warning 514: BRAM[0].RAMB18_Y1_RDCOUNT[2] unconnected pin in architecture.
Warning 515: BRAM[0].RAMB18_Y1_RDCOUNT[3] unconnected pin in architecture.
Warning 516: BRAM[0].RAMB18_Y1_RDCOUNT[4] unconnected pin in architecture.
Warning 517: BRAM[0].RAMB18_Y1_RDCOUNT[5] unconnected pin in architecture.
Warning 518: BRAM[0].RAMB18_Y1_RDCOUNT[6] unconnected pin in architecture.
Warning 519: BRAM[0].RAMB18_Y1_RDCOUNT[7] unconnected pin in architecture.
Warning 520: BRAM[0].RAMB18_Y1_RDCOUNT[8] unconnected pin in architecture.
Warning 521: BRAM[0].RAMB18_Y1_RDCOUNT[9] unconnected pin in architecture.
Warning 522: BRAM[0].RAMB18_Y1_RDCOUNT[10] unconnected pin in architecture.
Warning 523: BRAM[0].RAMB18_Y1_RDCOUNT[11] unconnected pin in architecture.
Warning 524: BRAM[0].RAMB18_Y1_FULL[0] unconnected pin in architecture.
Warning 525: BRAM[0].RAMB18_Y1_EMPTY[0] unconnected pin in architecture.
Warning 526: BRAM[0].RAMB18_Y1_ALMOSTFULL[0] unconnected pin in architecture.
Warning 527: BRAM[0].RAMB18_Y1_ALMOSTEMPTY[0] unconnected pin in architecture.
Warning 528: RAMBFIFO36E1[0].TSTWROS[0] unconnected pin in architecture.
Warning 529: RAMBFIFO36E1[0].TSTWROS[1] unconnected pin in architecture.
Warning 530: RAMBFIFO36E1[0].TSTWROS[2] unconnected pin in architecture.
Warning 531: RAMBFIFO36E1[0].TSTWROS[3] unconnected pin in architecture.
Warning 532: RAMBFIFO36E1[0].TSTWROS[4] unconnected pin in architecture.
Warning 533: RAMBFIFO36E1[0].TSTWROS[5] unconnected pin in architecture.
Warning 534: RAMBFIFO36E1[0].TSTWROS[6] unconnected pin in architecture.
Warning 535: RAMBFIFO36E1[0].TSTWROS[7] unconnected pin in architecture.
Warning 536: RAMBFIFO36E1[0].TSTWROS[8] unconnected pin in architecture.
Warning 537: RAMBFIFO36E1[0].TSTWROS[9] unconnected pin in architecture.
Warning 538: RAMBFIFO36E1[0].TSTWROS[10] unconnected pin in architecture.
Warning 539: RAMBFIFO36E1[0].TSTWROS[11] unconnected pin in architecture.
Warning 540: RAMBFIFO36E1[0].TSTWROS[12] unconnected pin in architecture.
Warning 541: RAMBFIFO36E1[0].TSTWRCNTOFF[0] unconnected pin in architecture.
Warning 542: RAMBFIFO36E1[0].TSTRDOS[0] unconnected pin in architecture.
Warning 543: RAMBFIFO36E1[0].TSTRDOS[1] unconnected pin in architecture.
Warning 544: RAMBFIFO36E1[0].TSTRDOS[2] unconnected pin in architecture.
Warning 545: RAMBFIFO36E1[0].TSTRDOS[3] unconnected pin in architecture.
Warning 546: RAMBFIFO36E1[0].TSTRDOS[4] unconnected pin in architecture.
Warning 547: RAMBFIFO36E1[0].TSTRDOS[5] unconnected pin in architecture.
Warning 548: RAMBFIFO36E1[0].TSTRDOS[6] unconnected pin in architecture.
Warning 549: RAMBFIFO36E1[0].TSTRDOS[7] unconnected pin in architecture.
Warning 550: RAMBFIFO36E1[0].TSTRDOS[8] unconnected pin in architecture.
Warning 551: RAMBFIFO36E1[0].TSTRDOS[9] unconnected pin in architecture.
Warning 552: RAMBFIFO36E1[0].TSTRDOS[10] unconnected pin in architecture.
Warning 553: RAMBFIFO36E1[0].TSTRDOS[11] unconnected pin in architecture.
Warning 554: RAMBFIFO36E1[0].TSTRDOS[12] unconnected pin in architecture.
Warning 555: RAMBFIFO36E1[0].TSTRDCNTOFF[0] unconnected pin in architecture.
Warning 556: RAMBFIFO36E1[0].TSTOFF[0] unconnected pin in architecture.
Warning 557: RAMBFIFO36E1[0].TSTIN[0] unconnected pin in architecture.
Warning 558: RAMBFIFO36E1[0].TSTIN[1] unconnected pin in architecture.
Warning 559: RAMBFIFO36E1[0].TSTIN[2] unconnected pin in architecture.
Warning 560: RAMBFIFO36E1[0].TSTIN[3] unconnected pin in architecture.
Warning 561: RAMBFIFO36E1[0].TSTIN[4] unconnected pin in architecture.
Warning 562: RAMBFIFO36E1[0].TSTFLAGIN[0] unconnected pin in architecture.
Warning 563: RAMBFIFO36E1[0].TSTCNT[0] unconnected pin in architecture.
Warning 564: RAMBFIFO36E1[0].TSTCNT[1] unconnected pin in architecture.
Warning 565: RAMBFIFO36E1[0].TSTCNT[2] unconnected pin in architecture.
Warning 566: RAMBFIFO36E1[0].TSTCNT[3] unconnected pin in architecture.
Warning 567: RAMBFIFO36E1[0].TSTCNT[4] unconnected pin in architecture.
Warning 568: RAMBFIFO36E1[0].TSTCNT[5] unconnected pin in architecture.
Warning 569: RAMBFIFO36E1[0].TSTCNT[6] unconnected pin in architecture.
Warning 570: RAMBFIFO36E1[0].TSTCNT[7] unconnected pin in architecture.
Warning 571: RAMBFIFO36E1[0].TSTCNT[8] unconnected pin in architecture.
Warning 572: RAMBFIFO36E1[0].TSTCNT[9] unconnected pin in architecture.
Warning 573: RAMBFIFO36E1[0].TSTCNT[10] unconnected pin in architecture.
Warning 574: RAMBFIFO36E1[0].TSTCNT[11] unconnected pin in architecture.
Warning 575: RAMBFIFO36E1[0].TSTCNT[12] unconnected pin in architecture.
Warning 576: RAMBFIFO36E1[0].TSTBRAMRST[0] unconnected pin in architecture.
Warning 577: RAMBFIFO36E1[0].REGCLKBU[0] unconnected pin in architecture.
Warning 578: RAMBFIFO36E1[0].REGCLKBL[0] unconnected pin in architecture.
Warning 579: RAMBFIFO36E1[0].REGCLKARDRCLKU[0] unconnected pin in architecture.
Warning 580: RAMBFIFO36E1[0].REGCLKARDRCLKL[0] unconnected pin in architecture.
Warning 581: RAMBFIFO36E1[0].INJECTDBITERR[0] unconnected pin in architecture.
Warning 582: RAMBFIFO36E1[0].INJECTSBITERR[0] unconnected pin in architecture.
Warning 583: RAMBFIFO36E1[0].WRERR[0] unconnected pin in architecture.
Warning 584: RAMBFIFO36E1[0].WRCOUNT[0] unconnected pin in architecture.
Warning 585: RAMBFIFO36E1[0].WRCOUNT[1] unconnected pin in architecture.
Warning 586: RAMBFIFO36E1[0].WRCOUNT[2] unconnected pin in architecture.
Warning 587: RAMBFIFO36E1[0].WRCOUNT[3] unconnected pin in architecture.
Warning 588: RAMBFIFO36E1[0].WRCOUNT[4] unconnected pin in architecture.
Warning 589: RAMBFIFO36E1[0].WRCOUNT[5] unconnected pin in architecture.
Warning 590: RAMBFIFO36E1[0].WRCOUNT[6] unconnected pin in architecture.
Warning 591: RAMBFIFO36E1[0].WRCOUNT[7] unconnected pin in architecture.
Warning 592: RAMBFIFO36E1[0].WRCOUNT[8] unconnected pin in architecture.
Warning 593: RAMBFIFO36E1[0].WRCOUNT[9] unconnected pin in architecture.
Warning 594: RAMBFIFO36E1[0].WRCOUNT[10] unconnected pin in architecture.
Warning 595: RAMBFIFO36E1[0].WRCOUNT[11] unconnected pin in architecture.
Warning 596: RAMBFIFO36E1[0].WRCOUNT[12] unconnected pin in architecture.
Warning 597: RAMBFIFO36E1[0].TSTOUT[0] unconnected pin in architecture.
Warning 598: RAMBFIFO36E1[0].TSTOUT[1] unconnected pin in architecture.
Warning 599: RAMBFIFO36E1[0].TSTOUT[2] unconnected pin in architecture.
Warning 600: RAMBFIFO36E1[0].TSTOUT[3] unconnected pin in architecture.
Warning 601: RAMBFIFO36E1[0].TSTOUT[4] unconnected pin in architecture.
Warning 602: RAMBFIFO36E1[0].SBITERR[0] unconnected pin in architecture.
Warning 603: RAMBFIFO36E1[0].RDERR[0] unconnected pin in architecture.
Warning 604: RAMBFIFO36E1[0].RDCOUNT[0] unconnected pin in architecture.
Warning 605: RAMBFIFO36E1[0].RDCOUNT[1] unconnected pin in architecture.
Warning 606: RAMBFIFO36E1[0].RDCOUNT[2] unconnected pin in architecture.
Warning 607: RAMBFIFO36E1[0].RDCOUNT[3] unconnected pin in architecture.
Warning 608: RAMBFIFO36E1[0].RDCOUNT[4] unconnected pin in architecture.
Warning 609: RAMBFIFO36E1[0].RDCOUNT[5] unconnected pin in architecture.
Warning 610: RAMBFIFO36E1[0].RDCOUNT[6] unconnected pin in architecture.
Warning 611: RAMBFIFO36E1[0].RDCOUNT[7] unconnected pin in architecture.
Warning 612: RAMBFIFO36E1[0].RDCOUNT[8] unconnected pin in architecture.
Warning 613: RAMBFIFO36E1[0].RDCOUNT[9] unconnected pin in architecture.
Warning 614: RAMBFIFO36E1[0].RDCOUNT[10] unconnected pin in architecture.
Warning 615: RAMBFIFO36E1[0].RDCOUNT[11] unconnected pin in architecture.
Warning 616: RAMBFIFO36E1[0].RDCOUNT[12] unconnected pin in architecture.
Warning 617: RAMBFIFO36E1[0].FULL[0] unconnected pin in architecture.
Warning 618: RAMBFIFO36E1[0].EMPTY[0] unconnected pin in architecture.
Warning 619: RAMBFIFO36E1[0].ECCPARITY[0] unconnected pin in architecture.
Warning 620: RAMBFIFO36E1[0].ECCPARITY[1] unconnected pin in architecture.
Warning 621: RAMBFIFO36E1[0].ECCPARITY[2] unconnected pin in architecture.
Warning 622: RAMBFIFO36E1[0].ECCPARITY[3] unconnected pin in architecture.
Warning 623: RAMBFIFO36E1[0].ECCPARITY[4] unconnected pin in architecture.
Warning 624: RAMBFIFO36E1[0].ECCPARITY[5] unconnected pin in architecture.
Warning 625: RAMBFIFO36E1[0].ECCPARITY[6] unconnected pin in architecture.
Warning 626: RAMBFIFO36E1[0].ECCPARITY[7] unconnected pin in architecture.
Warning 627: RAMBFIFO36E1[0].DBITERR[0] unconnected pin in architecture.
Warning 628: RAMBFIFO36E1[0].ALMOSTFULL[0] unconnected pin in architecture.
Warning 629: RAMBFIFO36E1[0].ALMOSTEMPTY[0] unconnected pin in architecture.
# Building complex block graph took 0.07 seconds (max_rss 15.8 MiB, delta_rss +1.6 MiB)
Error 1: 
Type: Blif file
File: /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/build/xc7/tests/dram/dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.eblif
Line: -1
Message: No non-blackbox models found. The main model must not be a blackbox.
# Load circuit
# Load circuit took 0.00 seconds (max_rss 15.8 MiB, delta_rss +0.0 MiB)
The entire flow of VPR took 0.16 seconds (max_rss 15.8 MiB)
xc7/tests/dram/CMakeFiles/dram_2_32x1d_bin.dir/build.make:133: recipe for target 'xc7/tests/dram/dram_2_32x1d/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top.net' failed
```

I think I got another race condition when building from `fd521c5`:
```
...
[ 33%] Generating dram_1_64m/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_io.place
Traceback (most recent call last):
  File "/tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/xc7/utils/prjxray_create_ioplace.py", line 94, in <module>
    main()
  File "/tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/xc7/utils/prjxray_create_ioplace.py", line 49, in main
    io_place.read_io_list_from_eblif(args.blif)
  File "/tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/utils/vpr_io_place.py", line 21, in read_io_list_from_eblif
    blif = eblif.parse_blif(eblif_file)
  File "/tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7/utils/eblif.py", line 69, in parse_blif
    assert len(data['inputs']) == 1
KeyError: 'inputs'
xc7/tests/dram/CMakeFiles/dram_1_64m_bin.dir/build.make:111: recipe for target 'xc7/tests/dram/dram_1_64m/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_io.place' failed
make[3]: *** [xc7/tests/dram/dram_1_64m/artix7-xc7a50t-basys3-roi-virt-xc7a50t-basys3-test/top_io.place] Error 1
CMakeFiles/Makefile2:73215: recipe for target 'xc7/tests/dram/CMakeFiles/dram_1_64m_bin.dir/all' failed
make[2]: *** [xc7/tests/dram/CMakeFiles/dram_1_64m_bin.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
...
```

--- a/common/cmake/devices.cmake
+++ b/common/cmake/devices.cmake
@@ -802,6 +802,7 @@ function(ADD_FPGA_TARGET)
   # Generate BLIF as start of vpr input.
   #
   set(OUT_EBLIF ${OUT_LOCAL}/${TOP}.eblif)
+  set(OUT_EBLIF_REL ${OUT_LOCAL_REL}/${TOP}.eblif)
   set(OUT_SYNTH_V ${OUT_LOCAL}/${TOP}_synth.v)
 
   set(SOURCE_FILES_DEPS "")
@@ -834,7 +835,7 @@ function(ADD_FPGA_TARGET)
       WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
       VERBATIM
     )
-    add_file_target(FILE ${OUT_SYNTH_V} GENERATED)
+    add_output_to_fpga_target(${NAME} EBLIF ${OUT_EBLIF_REL})
     add_output_to_fpga_target(${NAME} SYNTH_V
         ${OUT_LOCAL_REL}/${TOP}_synth.v)
   else()
@@ -845,6 +846,7 @@ function(ADD_FPGA_TARGET)
         ${CMAKE_COMMAND} -E make_directory ${OUT_LOCAL}
       COMMAND ${CMAKE_COMMAND} -E copy ${SOURCE_FILES} ${OUT_EBLIF}
       )
+    add_output_to_fpga_target(${NAME} EBLIF ${OUT_EBLIF_REL})
   endif()
 
   add_custom_target(${NAME}_eblif DEPENDS ${OUT_EBLIF})
@@ -853,17 +855,13 @@ function(ADD_FPGA_TARGET)
   set(OUT_ROUTE ${OUT_LOCAL}/${TOP}.route)
 
   set(VPR_DEPS "")
-  list(APPEND VPR_DEPS ${OUT_EBLIF})
 
   get_file_location(OUT_RRXML_VIRT_LOCATION ${OUT_RRXML_VIRT})
   get_file_location(OUT_RRXML_REAL_LOCATION ${OUT_RRXML_REAL})
   get_file_location(DEVICE_MERGED_FILE_LOCATION ${DEVICE_MERGED_FILE})
 
   foreach(SRC ${DEVICE_MERGED_FILE} ${OUT_RRXML_REAL})
-    get_file_location(SRC_LOCATION ${SRC})
-    get_file_target(SRC_TARGET ${SRC})
-    list(APPEND VPR_DEPS ${SRC_LOCATION})
-    list(APPEND VPR_DEPS ${SRC_TARGET})
+    append_file_dependency(VPR_DEPS ${SRC})
   endforeach()
 
   get_target_property_required(VPR env VPR)
@@ -908,44 +906,50 @@ function(ADD_FPGA_TARGET)
     ${SDC_ARG}
   )
   list(APPEND VPR_DEPS ${VPR} ${VPR_TARGET} ${QUIET_CMD} ${QUIET_CMD_TARGET})
+  append_file_dependency(VPR_DEPS ${OUT_EBLIF_REL})
 
   # Generate IO constraints file.
   # -------------------------------------------------------------------------
-  set(OUT_IO "")
   set(FIX_PINS_ARG "")
+
   if(NOT ${ADD_FPGA_TARGET_INPUT_IO_FILE} STREQUAL "")
     get_target_property_required(NO_PINS ${ARCH} NO_PINS)
     if(${NO_PINS})
       message(FATAL_ERROR "Arch ${ARCH} does not currently support pin constraints.")
     endif()
-    get_file_location(INPUT_IO_FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
-    get_file_target(INPUT_IO_FILE_TARGET ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     get_target_property_required(PLACE_TOOL ${ARCH} PLACE_TOOL)
     get_target_property_required(PYTHON3 env PYTHON3)
     get_target_property_required(PLACE_TOOL_CMD ${ARCH} PLACE_TOOL_CMD)
     get_target_property_required(PINMAP_FILE ${DEVICE} ${PACKAGE}_PINMAP)
-    get_file_location(PINMAP ${PINMAP_FILE})
-    get_file_target(PINMAP_TARGET ${PINMAP_FILE})
+
+
+    # Add complete dependency chain
+    set(IO_DEPS ${VPR_DEPS})
+    append_file_dependency(IO_DEPS ${ADD_FPGA_TARGET_INPUT_IO_FILE})
+    append_file_dependency(IO_DEPS ${PINMAP_FILE})
+
+
+    # Set variables for the string(CONFIGURE) below.
     set(OUT_IO ${OUT_LOCAL}/${TOP}_io.place)
+    set(OUT_IO_REL ${OUT_LOCAL_REL}/${TOP}_io.place)
+    get_file_location(INPUT_IO_FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
+    get_file_location(PINMAP ${PINMAP_FILE})
     string(CONFIGURE ${PLACE_TOOL_CMD} PLACE_TOOL_CMD_FOR_TARGET)
     separate_arguments(
       PLACE_TOOL_CMD_FOR_TARGET_LIST UNIX_COMMAND ${PLACE_TOOL_CMD_FOR_TARGET}
     )
+
     add_custom_command(
       OUTPUT ${OUT_IO}
-      DEPENDS
-        ${OUT_EBLIF}
-        ${INPUT_IO_FILE}
-        ${INPUT_IO_FILE_TARGET}
-        ${PINMAP}
-        ${PINMAP_TARGET}
-        ${VPR_DEPS}
+      DEPENDS ${IO_DEPS}
       COMMAND ${PLACE_TOOL_CMD_FOR_TARGET_LIST} --out ${OUT_IO}
       WORKING_DIRECTORY ${OUT_LOCAL}
     )
 
-    set(FIX_PINS_ARG --fix_pins ${OUT_IO})
+    add_output_to_fpga_target(${NAME} IO_PLACE ${OUT_IO_REL})
+    append_file_dependency(VPR_DEPS ${OUT_IO_REL})
 
+    set(FIX_PINS_ARG --fix_pins ${OUT_IO})
   endif()
 
   # Generate packing.
@@ -953,7 +957,7 @@ function(ADD_FPGA_TARGET)
   set(OUT_NET ${OUT_LOCAL}/${TOP}.net)
   add_custom_command(
     OUTPUT ${OUT_NET} ${OUT_LOCAL}/pack.log
-    DEPENDS ${OUT_EBLIF} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${VPR_DEPS}
     COMMAND ${VPR_CMD} --pack
     COMMAND
       ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log ${OUT_LOCAL}/pack.log
@@ -974,7 +978,7 @@ function(ADD_FPGA_TARGET)
   set(ECHO_OUT_NET ${OUT_LOCAL}/echo/${TOP}.net)
   add_custom_command(
     OUTPUT ${ECHO_OUT_NET}
-    DEPENDS ${OUT_EBLIF} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${VPR_DEPS}
     COMMAND ${CMAKE_COMMAND} -E make_directory ${OUT_LOCAL}/echo
     COMMAND cd ${OUT_LOCAL}/echo && ${VPR_CMD} --pack_verbosity 3 --echo_file on --pack
     COMMAND
@@ -986,7 +990,7 @@ function(ADD_FPGA_TARGET)
   set(OUT_PLACE ${OUT_LOCAL}/${TOP}.place)
   add_custom_command(
     OUTPUT ${OUT_PLACE}
-    DEPENDS ${OUT_NET} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${OUT_NET} ${VPR_DEPS}
     COMMAND ${VPR_CMD} ${FIX_PINS_ARG} --place
     COMMAND
       ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log
@@ -997,7 +1001,7 @@ function(ADD_FPGA_TARGET)
   set(ECHO_OUT_PLACE ${OUT_LOCAL}/echo/${TOP}.place)
   add_custom_command(
     OUTPUT ${ECHO_OUT_PLACE}
-    DEPENDS ${ECHO_OUT_NET} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${ECHO_OUT_NET} ${VPR_DEPS}
     COMMAND ${VPR_CMD} ${FIX_PINS_ARG} --echo_file on --place
     COMMAND
       ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/echo/vpr_stdout.log
@@ -1009,7 +1013,7 @@ function(ADD_FPGA_TARGET)
   # -------------------------------------------------------------------------
   add_custom_command(
     OUTPUT ${OUT_ROUTE}
-    DEPENDS ${OUT_PLACE} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${OUT_PLACE} ${VPR_DEPS}
     COMMAND ${VPR_CMD} --route
     COMMAND
       ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log
@@ -1023,7 +1027,7 @@ function(ADD_FPGA_TARGET)
   set(ECHO_ATOM_NETLIST_CLEANED ${OUT_LOCAL}/echo/atom_netlist.cleaned.echo.blif)
   add_custom_command(
     OUTPUT ${ECHO_ATOM_NETLIST_ORIG} ${ECHO_ATOM_NETLIST_CLEANED}
-    DEPENDS ${ECHO_OUT_PLACE} ${ECHO_OUT_IO} ${VPR_DEPS} ${ECHO_DIRECTORY_TARGET}
+    DEPENDS ${ECHO_OUT_PLACE} ${VPR_DEPS} ${ECHO_DIRECTORY_TARGET}
     COMMAND ${VPR_CMD} --echo_file on --route
     COMMAND
       ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/echo/vpr_stdout.log
@@ -1074,7 +1078,7 @@ function(ADD_FPGA_TARGET)
     set(OUT_FASM ${OUT_LOCAL}/${TOP}.fasm)
     add_custom_command(
       OUTPUT ${OUT_FASM}
-      DEPENDS ${OUT_ROUTE} ${OUT_PLACE} ${OUT_IO} ${VPR_DEPS} ${GENFASM_TARGET}
+      DEPENDS ${OUT_ROUTE} ${OUT_PLACE} ${VPR_DEPS} ${GENFASM_TARGET}
       COMMAND ${GENFASM_CMD}
       COMMAND
         ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log
@@ -1088,7 +1092,7 @@ function(ADD_FPGA_TARGET)
     set(OUT_HLC ${OUT_LOCAL}/${TOP}.hlc)
     add_custom_command(
       OUTPUT ${OUT_HLC}
-      DEPENDS ${OUT_ROUTE} ${OUT_PLACE} ${OUT_IO} ${VPR_DEPS} ${GENHLC_TARGET}
+      DEPENDS ${OUT_ROUTE} ${OUT_PLACE} ${VPR_DEPS} ${GENHLC_TARGET}
       COMMAND ${GENHLC_CMD}
       COMMAND
         ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log
@@ -1105,7 +1109,7 @@ function(ADD_FPGA_TARGET)
   set(OUT_POST_SYNTHESIS_BLIF ${OUT_LOCAL}/top_post_synthesis.blif)
   add_custom_command(
     OUTPUT ${OUT_ANALYSIS} ${OUT_POST_SYNTHESIS_V} ${OUT_POST_SYNTHESIS_BLIF}
-    DEPENDS ${OUT_ROUTE} ${OUT_IO} ${VPR_DEPS}
+    DEPENDS ${OUT_ROUTE} ${VPR_DEPS}
     COMMAND ${VPR_CMD} --analysis --gen_post_synthesis_netlist on
     COMMAND ${CMAKE_COMMAND} -E copy ${OUT_LOCAL}/vpr_stdout.log
         ${OUT_LOCAL}/analysis.log

